cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -134,13 +134,18 @@ public synchronized void create(NamespaceMeta metadata) throws Exception {
       authorizerInstantiator.get().enforce(instanceId, principal, Action.ADMIN);
     }
 
+    // store the meta first in the namespace store because namespacedlocationfactory need to look up location
+    // mapping from namespace config
```

Comment:
I think its because of how the NamespacedLocationFactory is designed/implemented.
It will query the namespace store, to lookup the mapping (to know whether to have custom mapping or not).